ingest/src/test/java/com/microsoft/azure/kusto/ingest/E2ETest.java

```diff
@@ -0,0 +1,252 @@
+            {
+                file = new File(resourcesPath, "dataset.csv");
+                rows = 10;
+                ingestionProperties = ingestionPropertiesWithoutMapping;
+            }
+        }, new TestDataItem() {
+            {
+                file = new File(resourcesPath, "dataset.csv.gz");
+                rows = 10;
+                ingestionProperties = ingestionPropertiesWithoutMapping;
+            }
+        }, new TestDataItem() {
+            {
+                file = new File(resourcesPath, "dataset.json");
+                rows = 2;
+                ingestionProperties = ingestionPropertiesWithMappingReference;
+            }
+        }, new TestDataItem() {
+            {
+                file = new File(resourcesPath, "dataset.json.gz");
+                rows = 2;
+                ingestionProperties = ingestionPropertiesWithMappingReference;
+            }
+        }, new TestDataItem() {
+            {
+                file = new File(resourcesPath, "dataset.json");
+                rows = 2;
+                ingestionProperties = ingestionPropertiesWithColumnMapping;
+                testOnstreamingIngestion = false; // streaming ingestion doesn't support inline mapping
+            }
+        }, new TestDataItem() {
+            {
+                file = new File(resourcesPath, "dataset.json.gz");
+                rows = 2;
+                ingestionProperties = ingestionPropertiesWithColumnMapping;
+                testOnstreamingIngestion = false; // streaming ingestion doesn't support inline mapping
+            }
+        });
+    }
+
+    private void assertRowCount(int expectedRowsCount) {
+        KustoOperationResult result = null;
+        int timeoutInSec = 100;
+        int actualRowsCount = 0;
+
+        while (timeoutInSec > 0) {
+            try {
+                Thread.sleep(5000);
+                timeoutInSec -= 5;
+
+                result = queryClient.execute(databaseName, String.format("%s | count", tableName));
+            } catch (Exception ex) {
+                continue;
+            }
+            KustoResultSetTable mainTableResult = result.getPrimaryResults();
+            mainTableResult.next();
+            actualRowsCount = mainTableResult.getInt(0) - currentCount;
+            if (actualRowsCount >= expectedRowsCount) {
+                break;
+            }
+        }
+        currentCount += actualRowsCount;
+        assertEquals(expectedRowsCount, actualRowsCount);
+    }
+
+    @Test
+    void testShowPrincipals() {
+        KustoOperationResult result = null;
+        boolean found = false;
+        try {
+            result = queryClient.execute(databaseName, String.format(".show database %s principals", databaseName));
+        } catch (Exception ex) {
+            Assertions.fail("Failed to execute show database principal command", ex);
+        }
+        KustoResultSetTable mainTableResultSet= result.getPrimaryResults();
+        while (mainTableResultSet.next()){
```

Comment:
I would add a comment here saying that this actually tests result parsing and not just simple exists